src/dirtyfields/dirtyfields.py

```diff
@@ -121,6 +124,8 @@ def get_dirty_fields(self, check_relationship=False, check_m2m=None, verbose=Fal
         return modified_fields
 
     def is_dirty(self, check_relationship=False, check_m2m=None):
+        if self.DISABLE_RESET_STATE_ON_M2M_CHANGED and self._m2m_dirty:
```

Comment:
Comment to say we return early if we are already aware there are dirty m2m fields? Maybe that's obvious.

Reply:
If we return early here, we will miss out on updating the modified fields for non-m2m fields. I think we should still call get dirty fields first before returning true (maybe pass in `check_m2m = False` to avoid calculating modified m2m fields).

Reply:
@RouganStriker we don't need to update it?
this is just returning true or false for if its dirty